src/app/cfg.rs

```diff
@@ -1,34 +1,43 @@
-use super::types::{SinkFn, UpdateFn, ViewFn, WindowEventsFn};
-use super::{builder::IntoAfterMount, MountType};
+use super::types::{UpdateFn, ViewFn};
 use crate::virtual_dom::IntoNodes;
-use std::marker::PhantomData;
 use std::rc::Rc;
 
 #[allow(clippy::module_name_repetitions)]
-pub struct AppInitCfg<Ms, Mdl, INodes, GMs, IAM: ?Sized>
+pub struct AppCfg<Ms, Mdl, INodes>
 where
     Ms: 'static,
     Mdl: 'static,
     INodes: IntoNodes<Ms>,
-    IAM: IntoAfterMount<Ms, Mdl, INodes, GMs>,
 {
-    pub mount_type: MountType,
-    pub into_after_mount: Box<IAM>,
-    pub phantom: PhantomData<(Ms, Mdl, INodes, GMs)>,
+    pub(crate) document: web_sys::Document,
+    // @TODO_B: get more context of use in order to understand how to remove Element mount point
+    mount_point: web_sys::Element,
+    pub(crate) update: UpdateFn<Ms, Mdl, INodes>,
+    pub(crate) view: ViewFn<Mdl, INodes>,
+    pub(crate) base_path: Rc<Vec<String>>,
 }
 
-#[allow(clippy::module_name_repetitions)]
-pub struct AppCfg<Ms, Mdl, INodes, GMs>
+impl<Ms, Mdl, INodes> AppCfg<Ms, Mdl, INodes>
 where
-    Ms: 'static,
-    Mdl: 'static,
     INodes: IntoNodes<Ms>,
 {
-    pub document: web_sys::Document,
-    pub mount_point: web_sys::Element,
-    pub update: UpdateFn<Ms, Mdl, INodes, GMs>,
-    pub sink: Option<SinkFn<Ms, Mdl, INodes, GMs>>,
-    pub view: ViewFn<Mdl, INodes>,
-    pub window_events: Option<WindowEventsFn<Ms, Mdl>>,
-    pub base_path: Rc<Vec<String>>,
+    pub(crate) fn new(
+        document: web_sys::Document,
+        mount_point: web_sys::Element,
+        update: UpdateFn<Ms, Mdl, INodes>,
+        view: ViewFn<Mdl, INodes>,
+        base_path: Rc<Vec<String>>,
+    ) -> Self {
+        Self {
+            document,
+            mount_point,
+            update,
+            view,
+            base_path,
+        }
+    }
+
+    pub(crate) fn mount_point(&self) -> &web_sys::Element {
+        &self.mount_point
+    }
```

Comment:
You could just make the field `pub(crate)` instead. I'm not sure that getters should be used unless you're trying to keep some state consistent. I think I said this earlier, too.